### PR TITLE
Update makefile with examples target and add versioning.

### PR DESCRIPTION
**Pull request by craigyam**
- Add test.examples makefile target
- Version the examples images

### CODE REVIEW

File: testing/generate_example_yaml.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+set -x
+set -o errexit
+
+# The release for which the yamls are being generated
+TAG=$1
```
Comment (gicaceres) [inline testing/generate_example_yaml.sh:21]: we should exit if TAG is empty

File: testing/kubernetes/test-kubernetes.sh [hunk 2]
```diff
@@ -18,6 +18,7 @@ set -x
 set -o errexit
 
 A8_TEST_SUITE=$1
+A8_RELEASE=$2
```
Comment (gicaceres) [inline testing/kubernetes/test-kubernetes.sh:21]: exit if empty

File: testing/docker/test-docker.sh [hunk 3]
```diff
@@ -18,6 +18,7 @@ set -x
 set -o errexit
 
 A8_TEST_SUITE=$1
+export A8_RELEASE=$2
```
Comment (gicaceres) [inline testing/docker/test-docker.sh]: We should exit if empty

**Review by gicaceres — CHANGES_REQUESTED** [review]

File: testing/generate_example_yaml.sh (hunk 1 above)
Comment (craigyam) [reply testing/generate_example_yaml.sh:21]: Done

**Review by gicaceres — APPROVED** [review]
LGTM

File: testing/kubernetes/test-kubernetes.sh (hunk 2 above)
Comment (craigyam) [reply testing/kubernetes/test-kubernetes.sh:21]: Validated this in the test suite check section.

**Comment by craigyam** [comment]
Need to update the bluemix.cfg as well.  Will generate it in the generate_example_yaml.sh script.

**Review by gicaceres — CHANGES_REQUESTED** [review]
Need to update bluemix examples

**Comment by craigyam** [comment]
Removed the yamls from the examples directory as they are generated by testing/generate_example_yaml.sh (make build.exampleapps).  They should still be included in the zip file generated by make release with the appropriate version number.

**Review by gicaceres — APPROVED** [review]
LGTM

**Review by rshriram — APPROVED** [review]
LGTM. Have you verified that when you tag a commit in master, and run make examples, you get the example YAML files in examples/folder?

**Comment by craigyam** [comment]
@rshriram I haven't tested it out in master, but I tested out the follow up push_dockerhub.sh changes (in a different fork) and verified that the yamls are generated with the version numbers included.

**Pull request merged**